sonar-csharp-plugin/src/test/java/org/sonar/plugins/csharp/CSharpSonarRulesDefinitionTest.java

```diff
@@ -84,4 +84,16 @@ public void test_security_standards_with_vulnerability() {
     assertThat(rule.type()).isEqualTo(RuleType.VULNERABILITY);
     assertThat(rule.securityStandards()).containsExactlyInAnyOrder("cwe:326", "owaspTop10:a3", "owaspTop10:a6");
   }
+
+  @Test
+  public void test_all_rules_have_status_set(){
+    CSharpSonarRulesDefinition definition = new CSharpSonarRulesDefinition();
+    RulesDefinition.Context context = new RulesDefinition.Context();
+    definition.define(context);
+    RulesDefinition.Repository repository = context.repository("csharpsquid");
+
+    for (RulesDefinition.Rule rule:repository.rules()) {
+      assertThat(rule.status()).isNotNull();
```

Comment:
another idea is to check that the status is one of the allowed values for SQ, to make sure it won't fail at startup

Reply:
I've initially thought at this and found out that the same test will hard fail if the status from `rules.xml` is not one of the enum values.